// aspose.cells/lib/aspose.cells.js
// Copyright (c) 2001-2024 Aspose Pty Ltd. All Rights Reserved.
// Powered by Aspose.Cells.
'use strict'
module.exports = require("./aspose.cells.nodejs");
exports = module.exports;

exports.AutoFillType = {
    Copy : 1,
    Default : 0,
    Formats : 3,
    Series : 2,
    Values : 4,
};

exports.AutoFitMergedCellsType = {
    None : 0,
    FirstLine : 1,
    LastLine : 2,
    EachLine : 3,
};

exports.AutoFitWrappedTextType = {
    Default : 0,
    Paragraph : 1,
};

exports.ChartTextDirectionType = {
    Horizontal : 0,
    Vertical : 1,
    Rotate90 : 2,
    Rotate270 : 3,
    Stacked : 4,
};

exports.PlotDataByType = {
    Row : 0,
    Column : 1,
};

exports.TickLabelAlignmentType = {
    Center : 0,
    Left : 1,
    Right : 2,
};

exports.ImageBinarizationMethod = {
    Threshold : 0,
    FloydSteinbergDithering : 1,
};

exports.CommentTitleType = {
    Cell : 0,
    Comment : 1,
    Note : 2,
    Reply : 3,
};

exports.AdjustFontSizeForRowType = {
    None : 0,
    EmptyRows : 1,
};

exports.SlideViewType = {
    View : 0,
    Print : 1,
};

exports.PresetThemeGradientType = {
    LightGradient : 0,
    TopSpotlight : 1,
    MediumGradient : 2,
    BottomSpotlight : 3,
    RadialGradient : 4,
};

exports.EquationCombiningCharacterType = {
    Unknown : -1,
    DotAbove : 0,
    Diaeresis : 1,
    ThreeDotsAbove : 2,
    CircumflexAccent : 3,
    Caron : 4,
    AcuteAccent : 5,
    GraveAccent : 6,
    Breve : 7,
    Tilde : 8,
    Overline : 9,
    DoubleOverline : 10,
    TopCurlyBracket : 11,
    BottomCurlyBracket : 12,
    LeftArrowAbove : 13,
    RightArrowAbove : 14,
    LeftRightArrowAbove : 15,
    LeftHarpoonAbove : 16,
    RightHarpoonAbove : 17,
    LeftwardsArrow : 18,
    RightwardsArrow : 19,
    LeftRightArrow : 20,
    LeftwardsDoubleArrow : 21,
    RightwardsDoubleArrow : 22,
    LeftRightDoubleArrow : 23,
};

exports.EquationMathematicalOperatorType = {
    Unknown : -1,
    ForAll : 0,
    Complement : 1,
    PartialDifferential : 2,
    Exists : 3,
    NotExists : 4,
    EmptySet : 5,
    Increment : 6,
    Nabla : 7,
    ElementOf : 8,
    NotAnElementOf : 9,
    SmallElementOf : 10,
    Contain : 11,
    NotContain : 12,
    SmallContain : 13,
    EndOfProof : 14,
    NaryProduct : 15,
    NaryCoproduct : 16,
    NarySummation : 17,
    LogicalAnd : 18,
    LogicalOr : 19,
    Intersection : 20,
    Union : 21,
    Integral : 22,
    DoubleIntegral : 23,
    TripleIntegral : 24,
    ContourIntegral : 25,
    SurfaceIntegral : 26,
    VolumeIntegral : 27,
    Clockwise : 28,
    ClockwiseContourIntegral : 29,
    AnticlockwiseContourIntegral : 30,
    NaryLogicalAnd : 31,
    NaryLogicalOr : 32,
    NaryIntersection : 33,
    NaryUnion : 34,
};

exports.EquationVerticalJustificationType = {
    Top : 0,
    Center : 1,
    Bottom : 2,
};

exports.LoadNumbersTableType = {
    OneTablePerSheet : 1,
    OverrideOtherTables : 2,
    TileTables : 4,
};

exports.OpenDocumentFormatVersionType = {
    None : 0,
    Odf11 : 1,
    Odf12 : 2,
    Odf13 : 3,
};

exports.PivotAreaType = {
    None : 0,
    Normal : 1,
    Data : 2,
    All : 3,
    Origin : 4,
    Button : 5,
    TopRight : 6,
};

exports.PivotTableSelectionType = {
    DataAndLabel : 0,
    DataOnly : 2,
    LabelOnly : 1,
};

exports.PivotFieldGroupType = {
    None : 0,
    DateTimeRange : 1,
    NumbericRange : 2,
    Discrete : 3,
};

exports.PivotItemPositionType = {
    Previous : 0,
    Next : 1,
    Custom : 2,
};

exports.ReserveMissingPivotItemType = {
    Default : 0,
    All : 1,
    None : 2,
};

exports.MergeEmptyTdType = {
    Default : 0,
    None : 1,
    MergeForcely : 2,
};

exports.PdfFontEncoding = {
    Identity : 0,
    AnsiPrefer : 1,
};

exports.DefaultEditLanguage = {
    Auto : 0,
    English : 1,
    CJK : 2,
};

exports.EmfRenderSetting = {
    EmfOnly : 0,
    EmfPlusPrefer : 1,
};

exports.XAdESType = {
    None : 0,
    XAdES : 1,
};

exports.ShapeAnchorType = {
    TwoCellAnchor : 0,
    OneCellAnchor : 1,
};

exports.EquationCharacterPositionType = {
    Top : 0,
    Bottom : 1,
};

exports.PdfCustomPropertiesExport = {
    None : 0,
    Standard : 1,
};

exports.CopyFormatType = {
    SameAsAbove : 0,
    SameAsBelow : 1,
    Clear : 2,
};

exports.AccessCacheOptions = {
    None : 0,
    All : -1,
    PositionAndSize : 3,
    CellsData : 9,
    CellDisplay : 1048710,
    GetFormula : 2097152,
    SetFormula : 4194304,
    CalculateFormula : 8388617,
    ConditionalFormatting : 16777225,
    Validation : 33554441,
};

exports.MemorySetting = {
    Normal : 0,
    MemoryPreference : 1,
};

exports.DataLabelShapeType = {
    Rect : 0,
    RoundRect : 1,
    Ellipse : 2,
    RightArrowCallout : 3,
    DownArrowCallout : 4,
    LeftArrowCallout : 5,
    UpArrowCallout : 6,
    WedgeRectCallout : 7,
    WedgeRoundRectCallout : 8,
    WedgeEllipseCallout : 9,
    LineCallout : 10,
    BentLineCallout : 11,
    LineWithAccentBarCallout : 12,
    BentLineWithAccentBarCallout : 13,
    Line : 14,
};

exports.MapChartRegionType = {
    Automatic : 0,
    DataOnly : 1,
    CountryRegionList : 2,
    World : 3,
};

exports.MapChartProjectionType = {
    Automatic : 0,
    Mercator : 1,
    Miller : 2,
    Albers : 3,
};

exports.MapChartLabelLayout = {
    BestFitOnly : 0,
    ShowAll : 1,
    None : 2,
};

exports.QuartileCalculationType = {
    Exclusive : 0,
    Inclusive : 1,
};

exports.ReflectionEffectType = {
    None : 0,
    Custom : 1,
    TightReflectionTouching : 2,
    HalfReflectionTouching : 3,
    FullReflectionTouching : 4,
    TightReflection4PtOffset : 5,
    HalfReflection4PtOffset : 6,
    FullReflection4PtOffset : 7,
    TightReflection8PtOffset : 8,
    HalfReflection8PtOffset : 9,
    FullReflection8PtOffset : 10,
};

exports.BevelPresetType = {
    None : 0,
    Angle : 1,
    ArtDeco : 2,
    Circle : 3,
    Convex : 4,
    CoolSlant : 5,
    Cross : 6,
    Divot : 7,
    HardEdge : 8,
    RelaxedInset : 9,
    Riblet : 10,
    Slope : 11,
    SoftRound : 12,
};

exports.PresetWordArtStyle = {
    WordArtStyle1 : 1,
    WordArtStyle2 : 2,
    WordArtStyle3 : 3,
    WordArtStyle4 : 4,
    WordArtStyle5 : 5,
    WordArtStyle6 : 6,
    WordArtStyle7 : 7,
    WordArtStyle8 : 8,
    WordArtStyle9 : 9,
    WordArtStyle10 : 10,
    WordArtStyle11 : 11,
    WordArtStyle12 : 12,
    WordArtStyle13 : 13,
    WordArtStyle14 : 14,
    WordArtStyle15 : 15,
    WordArtStyle16 : 16,
    WordArtStyle17 : 17,
    WordArtStyle18 : 18,
    WordArtStyle19 : 19,
    WordArtStyle20 : 20,
};

exports.ExternalLinkType = {
    DDELink : 0,
    External : 1,
};

exports.FontSourceType = {
    FontFile : 0,
    FontsFolder : 1,
    MemoryFont : 2,
};

exports.NameScopeType = {
    All : 0,
    Workbook : 1,
    Worksheet : 2,
};

exports.PasteOperationType = {
    None : 0,
    Add : 1,
    Subtract : 2,
    Multiply : 3,
    Divide : 4,
};

exports.SqlScriptOperatorType = {
    Insert : 0,
    Update : 1,
    Delete : 2,
};

exports.OdsCellFieldType = {
    Date : 0,
    SheetName : 1,
    Title : 2,
};

exports.OdsGeneratorType = {
    LibreOffice : 0,
    OpenOffice : 1,
};

exports.OdsPageBackgroundGraphicPositionType = {
    TopLeft : 0,
    TopCenter : 1,
    TopRight : 2,
    CenterLeft : 3,
    CenterCenter : 4,
    CenterRight : 5,
    BottomLeft : 6,
    BottomCenter : 7,
    BottomRight : 8,
};

exports.OdsPageBackgroundGraphicType = {
    Position : 0,
    Area : 1,
    Tile : 2,
};

exports.OdsPageBackgroundType = {
    None : 0,
    Color : 1,
    Graphic : 2,
};

exports.PageLayoutAlignmentType = {
    Bottom : 0,
    Center : 1,
    Left : 2,
    Right : 3,
    Top : 4,
};

exports.JsonExportHyperlinkType = {
    DisplayString : 0,
    Address : 1,
    HtmlString : 2,
};

exports.PaneStateType = {
    Frozen : 0,
    FrozenSplit : 1,
    Split : 2,
    Normal : 3,
};

exports.ResourceLoadingType = {
    Default : 0,
    Skip : 1,
    UserProvided : 2,
};

exports.OoxmlCompliance = {
    Ecma376_2006 : 0,
    Iso29500_2008_Strict : 1,
};

exports.OoxmlCompressionType = {
    Level1 : 1,
    Level2 : 2,
    Level3 : 3,
    Level4 : 4,
    Level5 : 5,
    Level6 : 6,
    Level7 : 7,
    Level8 : 8,
    Level9 : 9,
};

exports.TargetModeType = {
    External : 0,
    FilePath : 1,
    Email : 2,
    CellReference : 3,
};

exports.WebExtensionStoreType = {
    OMEX : 0,
    SPCatalog : 1,
    SPApp : 2,
    Exchange : 3,
    FileSystem : 4,
    Registry : 5,
    ExCatalog : 6,
};

exports.RenameStrategy = {
    Exception : 0,
    Digit : 1,
    Letter : 2,
};

exports.HtmlFormatHandlingType = {
    All : 0,
    None : 1,
    Rtf : 2,
};

exports.PdfOptimizationType = {
    Standard : 0,
    MinimumSize : 1,
};

exports.CellsUnitType = {
    Pixel : 1,
    Point : 2,
    Inch : 4,
    Cm : 6,
    Character : 7,
};

exports.MetadataType = {
    Encryption : 1,
    Decryption : 2,
    Document_Properties : 4,
};

exports.CellValueFormatStrategy = {
    None : 0,
    CellStyle : 1,
    DisplayStyle : 2,
    DisplayString : 3,
};

exports.ActiveXPersistenceType = {
    PropertyBag : 0,
    Storage : 1,
    Stream : 2,
    StreamInit : 3,
};

exports.DropButtonStyle = {
    Plain : 0,
    Arrow : 1,
    Ellipsis : 2,
    Reduce : 3,
};

exports.ShowDropButtonType = {
    Never : 0,
    Focus : 1,
    Always : 2,
};

exports.ControlBorderType = {
    None : 0,
    Single : 1,
};

exports.InputMethodEditorMode = {
    NoControl : 0,
    On : 1,
    Off : 2,
    Disable : 3,
    Hiragana : 4,
    Katakana : 5,
    KatakanaHalf : 6,
    AlphaFull : 7,
    Alpha : 8,
    HangulFull : 9,
    Hangul : 10,
    HanziFull : 11,
    Hanzi : 12,
};

exports.ControlCaptionAlignmentType = {
    Left : 0,
    Right : 1,
};

exports.ControlType = {
    CommandButton : 0,
    ComboBox : 1,
    CheckBox : 2,
    ListBox : 3,
    TextBox : 4,
    SpinButton : 5,
    RadioButton : 6,
    Label : 7,
    Image : 8,
    ToggleButton : 9,
    ScrollBar : 10,
    BarCode : 11,
    Unknown : 12,
};

exports.ControlListStyle = {
    Plain : 0,
    Option : 1,
};

exports.ControlMatchEntryType = {
    FirstLetter : 0,
    Complete : 1,
    None : 2,
};

exports.ControlMousePointerType = {
    Default : 0,
    Arrow : 1,
    Cross : 2,
    IBeam : 3,
    SizeNESW : 6,
    SizeNS : 7,
    SizeNWSE : 8,
    SizeWE : 9,
    UpArrow : 10,
    HourGlass : 11,
    NoDrop : 12,
    AppStarting : 13,
    Help : 14,
    SizeAll : 15,
    Custom : 99,
};

exports.ControlPictureAlignmentType = {
    TopLeft : 0,
    TopRight : 1,
    Center : 2,
    BottomLeft : 3,
    BottomRight : 4,
};

exports.ControlPicturePositionType = {
    LeftTop : 131072,
    LeftCenter : 327683,
    LeftBottom : 524294,
    RightTop : 2,
    RightCenter : 196613,
    RightBottom : 393224,
    AboveLeft : 393216,
    AboveCenter : 458753,
    AboveRight : 524290,
    BelowLeft : 6,
    BelowCenter : 65543,
    BelowRight : 131080,
    Center : 262148,
};

exports.ControlPictureSizeMode = {
    Clip : 0,
    Stretch : 1,
    Zoom : 3,
};

exports.ControlScrollBarType = {
    None : 0,
    Horizontal : 1,
    BarsVertical : 2,
    BarsBoth : 3,
};

exports.ControlScrollOrientation = {
    Auto : 3,
    Vertical : 0,
    Horizontal : 1,
};

exports.ControlSpecialEffectType = {
    Flat : 0,
    Raised : 1,
    Sunken : 2,
    Etched : 3,
    Bump : 6,
};

exports.CalculationPrecisionStrategy = {
    None : 0,
    Round : 1,
    Decimal : 2,
};

exports.BulletType = {
    None : 0,
    Character : 1,
    Picture : 2,
    AutoNumbered : 3,
};

exports.EquationDelimiterShapeType = {
    Centered : 0,
    Match : 1,
};

exports.EquationNodeType = {
    UnKnow : 0,
    Text : 1,
    Base : 100,
    Denominator : 101,
    Numerator : 102,
    FunctionName : 103,
    Subscript : 104,
    Superscript : 105,
    Degree : 106,
    MatrixRow : 107,
    EquationParagraph : 200,
    MathematicalEquation : 201,
    FractionEquation : 202,
    FunctionEquation : 203,
    DelimiterEquation : 204,
    NaryEquation : 205,
    RadicalEquation : 206,
    SuperscriptEquation : 207,
    SubscriptEquation : 208,
    SubSupEquation : 209,
    PreSubSupEquation : 210,
    AccentEquation : 211,
    BarEquation : 212,
    BorderBoxEquation : 213,
    BoxEquation : 214,
    GroupCharacterEquation : 215,
    MatrixEquation : 216,
};

exports.EquationFractionType = {
    Bar : 0,
    NoBar : 1,
    Linear : 2,
    Skewed : 3,
};

exports.EquationHorizontalJustificationType = {
    Center : 0,
    CenterGroup : 1,
    Left : 2,
    Right : 3,
};

exports.EquationLimitLocationType = {
    UndOvr : 0,
    SubSup : 1,
};

exports.LineSpaceSizeType = {
    Percentage : 0,
    Points : 1,
};

exports.TextAutonumberScheme = {
    None : 0,
    AlphaLcParenBoth : 1,
    AlphaLcParenR : 2,
    AlphaLcPeriod : 3,
    AlphaUcParenBoth : 4,
    AlphaUcParenR : 5,
    AlphaUcPeriod : 6,
    Arabic1Minus : 7,
    Arabic2Minus : 8,
    ArabicDbPeriod : 9,
    ArabicDbPlain : 10,
    ArabicParenBoth : 11,
    ArabicParenR : 12,
    ArabicPeriod : 13,
    ArabicPlain : 14,
    CircleNumDbPlain : 15,
    CircleNumWdBlackPlain : 16,
    CircleNumWdWhitePlain : 17,
    Ea1ChsPeriod : 18,
    Ea1ChsPlain : 19,
    Ea1ChtPeriod : 20,
    Ea1ChtPlain : 21,
    Ea1JpnChsDbPeriod : 22,
    Ea1JpnKorPeriod : 23,
    Ea1JpnKorPlain : 24,
    Hebrew2Minus : 25,
    HindiAlpha1Period : 26,
    HindiAlphaPeriod : 27,
    HindiNumParenR : 28,
    HindiNumPeriod : 29,
    RomanLcParenBoth : 30,
    RomanLcParenR : 31,
    RomanLcPeriod : 32,
    RomanUcParenBoth : 33,
    RomanUcParenR : 34,
    RomanUcPeriod : 35,
    ThaiAlphaParenBoth : 36,
    ThaiAlphaParenR : 37,
    ThaiAlphaPeriod : 38,
    ThaiNumParenBoth : 39,
    ThaiNumParenR : 40,
    ThaiNumPeriod : 41,
};

exports.TextFontAlignType = {
    Automatic : 0,
    Bottom : 1,
    Baseline : 2,
    Center : 3,
    Top : 4,
};

exports.TextNodeType = {
    TextRun : 0,
    TextParagraph : 1,
    Equation : 2,
};

exports.TextTabAlignmentType = {
    Center : 0,
    Decimal : 1,
    Left : 2,
    Right : 3,
};

exports.TextVerticalType = {
    Vertical : 0,
    Horizontal : 1,
    VerticalLeftToRight : 2,
    Vertical90 : 3,
    Vertical270 : 4,
    Stacked : 5,
    StackedRightToLeft : 6,
};

exports.TxtLoadStyleStrategy = {
    None : 0,
    BuiltIn : 1,
    ExactFormat : 2,
};

exports.ShapeLockType = {
    Group : 0,
    AdjustHandles : 1,
    Text : 2,
    Points : 3,
    Crop : 4,
    Selection : 5,
    Move : 6,
    AspectRatio : 7,
    Rotation : 8,
    Ungroup : 9,
    Resize : 10,
    ShapeType : 11,
    Arrowhead : 12,
};

exports.ConnectionDataSourceType = {
    ODBCBasedSource : 1,
    DAOBasedSource : 2,
    FileBasedDataBaseSource : 3,
    WebQuery : 4,
    OLEDBBasedSource : 5,
    TextBasedSource : 6,
    ADORecordSet : 7,
    DSP : 8,
    OLEDBDataModel : 100,
    DataFeedDataModel : 101,
    WorksheetDataModel : 102,
    Table : 102,
    TextDataModel : 103,
    Unknown : 255,
};

exports.ConnectionParameterType = {
    Cell : 0,
    Prompt : 1,
    Value : 2,
};

exports.CredentialsMethodType = {
    Integrated : 0,
    None : 1,
    Prompt : 2,
    Stored : 3,
};

exports.OLEDBCommandType = {
    None : 0,
    CubeName : 1,
    SqlStatement : 2,
    TableName : 3,
    DefaultInformation : 4,
    WebBasedList : 5,
};

exports.ReConnectionMethodType = {
    Required : 1,
    Always : 2,
    Never : 3,
};

exports.SqlDataType = {
    SqlUnsignedOffset : -22,
    SqlSignedOffset : -20,
    SqlGUID : -11,
    SqlWLongVarchar : -10,
    SqlWVarchar : -9,
    SqlWChar : -8,
    SqlBit : -7,
    SqlTinyInt : -6,
    SqlBigInt : -5,
    SqlLongVarBinary : -4,
    SqlVarBinary : -3,
    SqlBinary : -2,
    SqlLongVarChar : -1,
    SqlUnknownType : 0,
    SqlChar : 1,
    SqlNumeric : 2,
    SqlDecimal : 3,
    SqlInteger : 4,
    SqlSmallInt : 5,
    SqlFloat : 6,
    SqlReal : 7,
    SqlDouble : 8,
    SqlTypeDate : 9,
    SqlTypeTime : 10,
    SqlTypeTimestamp : 11,
    SqlVarChar : 12,
    SqlIntervalYear : 101,
    SqlIntervalMonth : 102,
    SqlIntervalDay : 103,
    SqlIntervalHour : 104,
    SqlIntervalMinute : 105,
    SqlIntervalSecond : 106,
    SqlIntervalYearToMonth : 107,
    SqlIntervalDayToHour : 108,
    SqlIntervalDayToMinute : 109,
    SqlIntervalDayToSecond : 110,
    SqlIntervalHourToMinute : 111,
    SqlIntervalHourToSecond : 112,
    SqlIntervalMinuteToSecond : 113,
};

exports.LoadDataFilterOptions = {
    None : 0,
    All : 2147483647,
    CellBlank : 1,
    CellString : 2,
    CellNumeric : 4,
    CellError : 8,
    CellBool : 16,
    CellValue : 31,
    Formula : 32,
    CellData : 67108927,
    Chart : 256,
    Shape : 402653696,
    Drawing : 402653952,
    MergedArea : 1024,
    ConditionalFormatting : 2048,
    DataValidation : 4096,
    PivotTable : 8192,
    Table : 16384,
    Hyperlinks : 32768,
    SheetSettings : 65536,
    SheetData : 403701759,
    BookSettings : 1048576,
    Settings : 1114112,
    XmlMap : 2097152,
    Structure : 4194304,
    Document_Properties : 8388608,
    DefinedNames : 16777216,
    VBA : 33554432,
    Style : 67108864,
    Picture : 134217728,
    OleObject : 268435456,
    Revision : 536870912,
};

exports.FillType = {
    Automatic : 0,
    None : 1,
    Solid : 2,
    Gradient : 3,
    Texture : 4,
    Pattern : 5,
    Group : 6,
};

exports.DataBarAxisPosition = {
    Automatic : 0,
    Midpoint : 1,
    None : 2,
};

exports.DataBarBorderType = {
    None : 0,
    Solid : 1,
};

exports.DataBarFillType = {
    Solid : 0,
    Gradient : 1,
};

exports.DataBarNegativeColorType = {
    Color : 0,
    SameAsPositive : 1,
};

exports.PivotConditionFormatRuleType = {
    None : 0,
    All : 1,
    Row : 2,
    Column : 3,
};

exports.RevisionActionType = {
    Add : 0,
    Delete : 1,
    DeleteColumn : 2,
    DeleteRow : 3,
    InsertColumn : 4,
    InsertRow : 5,
};

exports.RevisionType = {
    CustomView : 0,
    DefinedName : 1,
    ChangeCells : 2,
    AutoFormat : 3,
    MergeConflict : 4,
    Comment : 5,
    Format : 6,
    InsertSheet : 7,
    MoveCells : 8,
    Undo : 9,
    QueryTable : 10,
    InsertDelete : 11,
    RenameSheet : 12,
    Unknown : 13,
};

exports.HtmlExportDataOptions = {
    Table : 1,
    All : 255,
};

exports.HtmlLinkTargetType = {
    Blank : 0,
    Parent : 1,
    Self : 2,
    Top : 3,
};

exports.TxtValueQuoteType = {
    Normal : 0,
    Always : 1,
    Minimum : 2,
    Never : 3,
};

exports.HtmlHiddenRowDisplayType = {
    Hidden : 0,
    Remove : 1,
};

exports.HtmlHiddenColDisplayType = {
    Hidden : 0,
    Remove : 1,
};

exports.SlicerCacheCrossFilterType = {
    None : 0,
    ShowItemsWithDataAtTop : 1,
    ShowItemsWithNoData : 2,
};

exports.SlicerCacheItemSortType = {
    Natural : 0,
    Ascending : 1,
    Descending : 2,
};

exports.SlicerStyleType = {
    SlicerStyleLight1 : 0,
    SlicerStyleLight2 : 1,
    SlicerStyleLight3 : 2,
    SlicerStyleLight4 : 3,
    SlicerStyleLight5 : 4,
    SlicerStyleLight6 : 5,
    SlicerStyleOther1 : 6,
    SlicerStyleOther2 : 7,
    SlicerStyleDark1 : 8,
    SlicerStyleDark2 : 9,
    SlicerStyleDark3 : 10,
    SlicerStyleDark4 : 11,
    SlicerStyleDark5 : 12,
    SlicerStyleDark6 : 13,
    Custom : 14,
};

exports.NumberCategoryType = {
    General : 0,
    Text : 1,
    Number : 2,
    Date : 3,
    Time : 4,
    Fraction : 5,
    Scientific : 6,
};

exports.TextCapsType = {
    None : 0,
    All : 1,
    Small : 2,
};

exports.TextStrikeType = {
    Single : 0,
    Double : 1,
    None : 2,
};

exports.UpdateLinksType = {
    UserSet : 0,
    Never : 1,
    Always : 2,
};

exports.VbaModuleType = {
    Procedural : 0,
    Document : 1,
    Class : 2,
    Designer : 3,
};

exports.VbaProjectReferenceType = {
    Registered : 0,
    Control : 1,
    Project : 2,
};

exports.WarningType = {
    FontSubstitution : 0,
    DuplicateDefinedName : 1,
    UnsupportedFileFormat : 2,
    InvalidTextOfDefinedName : 3,
    InvalidFontName : 4,
    InvalidAutoFilterRange : 5,
    IO : 6,
    Limitation : 7,
    InvalidData : 8,
    Formula : 9,
    InvalidOperator : 10,
};

exports.HtmlCrossType = {
    Default : 0,
    MSExport : 1,
    Cross : 2,
    CrossHideRight : 3,
    FitToCell : 4,
};

exports.ColorDepth = {
    Default : 0,
    Format1bpp : 1,
    Format4bpp : 4,
    Format8bpp : 8,
    Format24bpp : 24,
    Format32bpp : 32,
};

exports.DrawObjectEnum = {
    Image : 0,
    Cell : 1,
};

exports.GridlineType = {
    Dotted : 0,
    Hair : 1,
};

exports.PrintingPageType = {
    Default : 0,
    IgnoreBlank : 1,
    IgnoreStyle : 2,
};

exports.TextCrossType = {
    Default : 1,
    CrossKeep : 2,
    CrossOverride : 3,
    StrictInCell : 4,
};

exports.TiffCompression = {
    CompressionNone : 0,
    CompressionRle : 1,
    CompressionLZW : 2,
    CompressionCCITT3 : 3,
    CompressionCCITT4 : 4,
};

exports.ParameterType = {
    Reference : 0,
    Value : 1,
    Array : 2,
};

exports.ErrorBarDisplayType = {
    Both : 0,
    Minus : 1,
    None : 2,
    Plus : 3,
};

exports.FormatSetType = {
    None : 0,
    IsGradientSet : 1,
    IsTextureSet : 2,
    IsPatternSet : 3,
};

exports.LabelPositionType = {
    Center : 0,
    InsideBase : 1,
    InsideEnd : 2,
    OutsideEnd : 3,
    Above : 4,
    Below : 5,
    Left : 6,
    Right : 7,
    BestFit : 8,
    Moved : 9,
};

exports.MirrorType = {
    None : 0,
    Horizonal : 1,
    Vertical : 2,
    Both : 3,
};

exports.ShapePathType = {
    LineTo : 0,
    CubicBezierCurveTo : 1,
    MoveTo : 2,
    Close : 3,
    End : 4,
    Escape : 5,
    ArcTo : 6,
    Unknown : 7,
};

exports.ErrorCheckType = {
    EvaluationError : 1,
    Calc : 1,
    EmptyCellRef : 2,
    NumberStoredAsText : 4,
    TextNumber : 4,
    InconsistRange : 8,
    InconsistFormula : 16,
    TwoDigitTextYear : 32,
    TextDate : 32,
    UnlockedFormula : 64,
    UnproctedFormula : 64,
    TableDataValidation : 128,
    Validation : 128,
    CalculatedColumn : 129,
};

exports.LoadFormat = {
    Auto : 0,
    Csv : 1,
    CSV : 1,
    Xlsx : 6,
    Tsv : 11,
    TSV : 11,
    TabDelimited : 11,
    Html : 12,
    MHtml : 13,
    Ods : 14,
    ODS : 14,
    Excel97To2003 : 5,
    SpreadsheetML : 15,
    Xlsb : 16,
    Ots : 31,
    Numbers : 56,
    Fods : 59,
    FODS : 59,
    Sxc : 60,
    SXC : 60,
    Xml : 51,
    Epub : 52,
    Azw3 : 53,
    Unknown : 255,
    Image : 254,
    Json : 513,
};

exports.PivotConditionFormatScopeType = {
    Data : 0,
    Field : 1,
    Selection : 2,
};

exports.PivotFilterType = {
    CaptionBeginsWith : 0,
    CaptionBetween : 1,
    CaptionContains : 2,
    CaptionEndsWith : 3,
    CaptionEqual : 4,
    CaptionGreaterThan : 5,
    CaptionGreaterThanOrEqual : 6,
    CaptionLessThan : 7,
    CaptionLessThanOrEqual : 8,
    CaptionNotBeginsWith : 9,
    CaptionNotBetween : 10,
    CaptionNotContains : 11,
    CaptionNotEndsWith : 12,
    CaptionNotEqual : 13,
    Count : 14,
    DateBetween : 15,
    DateEqual : 16,
    DateNewerThan : 17,
    DateNewerThanOrEqual : 18,
    DateNotBetween : 19,
    DateNotEqual : 20,
    DateOlderThan : 21,
    DateOlderThanOrEqual : 22,
    LastMonth : 23,
    LastQuarter : 24,
    LastWeek : 25,
    LastYear : 26,
    M1 : 27,
    M2 : 28,
    M3 : 29,
    M4 : 30,
    M5 : 31,
    M6 : 32,
    M7 : 33,
    M8 : 34,
    M9 : 35,
    M10 : 36,
    M11 : 37,
    M12 : 38,
    NextMonth : 39,
    NextQuarter : 40,
    NextWeek : 41,
    NextYear : 42,
    Percent : 43,
    Q1 : 44,
    Q2 : 45,
    Q3 : 46,
    Q4 : 47,
    Sum : 48,
    ThisMonth : 49,
    ThisQuarter : 50,
    ThisWeek : 51,
    ThisYear : 52,
    Today : 53,
    Tomorrow : 54,
    Unknown : 55,
    ValueBetween : 56,
    ValueEqual : 57,
    ValueGreaterThan : 58,
    ValueGreaterThanOrEqual : 59,
    ValueLessThan : 60,
    ValueLessThanOrEqual : 61,
    ValueNotBetween : 62,
    ValueNotEqual : 63,
    YearToDate : 64,
    Yesterday : 65,
};

exports.PivotMissingItemLimitType = {
    Automatic : 0,
    Max : 1,
    None : 2,
};

exports.SortOnType = {
    Value : 0,
    CellColor : 1,
    FontColor : 2,
    Icon : 3,
};

exports.FormatConditionValueType = {
    Formula : 0,
    Max : 1,
    Min : 2,
    Number : 3,
    Percent : 4,
    Percentile : 5,
    AutomaticMax : 6,
    AutomaticMin : 7,
};

exports.IconSetType = {
    Arrows3 : 0,
    ArrowsGray3 : 1,
    Flags3 : 2,
    Signs3 : 3,
    Symbols3 : 4,
    Symbols32 : 5,
    TrafficLights31 : 6,
    TrafficLights32 : 7,
    Arrows4 : 8,
    ArrowsGray4 : 9,
    Rating4 : 10,
    RedToBlack4 : 11,
    TrafficLights4 : 12,
    Arrows5 : 13,
    ArrowsGray5 : 14,
    Quarters5 : 15,
    Rating5 : 16,
    Stars3 : 17,
    Boxes5 : 18,
    Triangles3 : 19,
    None : 20,
    CustomSet : 21,
    Smilies3 : 22,
    ColorSmilies3 : 23,
};

exports.TimePeriodType = {
    Today : 0,
    Yesterday : 1,
    Tomorrow : 2,
    Last7Days : 3,
    ThisMonth : 4,
    LastMonth : 5,
    NextMonth : 6,
    ThisWeek : 7,
    LastWeek : 8,
    NextWeek : 9,
    ThisYear : 10,
    LastYear : 11,
    NextYear : 12,
};

exports.ContentDisposition = {
    Inline : 0,
    Attachment : 1,
};

exports.HeaderFooterCommandType = {
    Text : 0,
    CurrentPage : 1,
    Pagecount : 2,
    CurrentDate : 3,
    CurrentTime : 4,
    SheetName : 5,
    FileName : 6,
    FilePath : 7,
    Picture : 8,
};

exports.LightRigDirectionType = {
    Bottom : 0,
    BottomLeft : 1,
    BottomRight : 2,
    Left : 3,
    Right : 4,
    Top : 5,
    TopLeft : 6,
    TopRight : 7,
};

exports.MsoArrowheadLength = {
    Short : 0,
    Medium : 1,
    Long : 2,
};

exports.MsoArrowheadWidth = {
    Narrow : 0,
    Medium : 1,
    Wide : 2,
};

exports.TextOverflowType = {
    Clip : 0,
    Ellipsis : 1,
    Overflow : 2,
};

exports.PasteType = {
    All : 0,
    Default : 1,
    AllExceptBorders : 2,
    DefaultExceptBorders : 3,
    ColumnWidths : 4,
    RowHeights : 5,
    Comments : 6,
    Formats : 7,
    Formulas : 8,
    FormulasAndNumberFormats : 9,
    Validation : 10,
    Values : 11,
    ValuesAndFormats : 12,
    ValuesAndNumberFormats : 13,
};

exports.PivotFieldDataDisplayFormat = {
    Normal : 0,
    DifferenceFrom : 1,
    PercentageOf : 2,
    PercentageDifferenceFrom : 3,
    RunningTotalIn : 4,
    PercentageOfRow : 5,
    PercentageOfColumn : 6,
    PercentageOfTotal : 7,
    Index : 8,
    PercentageOfParentRowTotal : 9,
    PercentageOfParentColumnTotal : 10,
    PercentageOfParentTotal : 11,
    PercentageOfRunningTotalIn : 12,
    RankSmallestToLargest : 13,
    RankLargestToSmallest : 14,
};

exports.PivotFieldSubtotalType = {
    None : 0,
    Automatic : 1,
    Sum : 2,
    Count : 4,
    Average : 8,
    Max : 16,
    Min : 32,
    Product : 64,
    CountNums : 128,
    Stdev : 256,
    Stdevp : 512,
    Var : 1024,
    Varp : 2048,
};

exports.PivotFieldType = {
    Undefined : 0,
    Row : 1,
    Column : 2,
    Page : 4,
    Data : 8,
};

exports.PivotGroupByType = {
    RangeOfValues : 0,
    Numbers : 0,
    Seconds : 1,
    Minutes : 2,
    Hours : 3,
    Days : 4,
    Months : 5,
    Quarters : 6,
    Years : 7,
};

exports.PivotItemPosition = {
    Previous : 0,
    Next : 1,
    Custom : 2,
};

exports.PivotTableAutoFormatType = {
    None : 0,
    Classic : 1,
    Report1 : 2,
    Report2 : 3,
    Report3 : 4,
    Report4 : 5,
    Report5 : 6,
    Report6 : 7,
    Report7 : 8,
    Report8 : 9,
    Report9 : 10,
    Report10 : 11,
    Table1 : 12,
    Table2 : 13,
    Table3 : 14,
    Table4 : 15,
    Table5 : 16,
    Table6 : 17,
    Table7 : 18,
    Table8 : 19,
    Table9 : 20,
    Table10 : 21,
};

exports.PivotTableStyleType = {
    None : 0,
    PivotTableStyleLight1 : 1,
    PivotTableStyleLight2 : 2,
    PivotTableStyleLight3 : 3,
    PivotTableStyleLight4 : 4,
    PivotTableStyleLight5 : 5,
    PivotTableStyleLight6 : 6,
    PivotTableStyleLight7 : 7,
    PivotTableStyleLight8 : 8,
    PivotTableStyleLight9 : 9,
    PivotTableStyleLight10 : 10,
    PivotTableStyleLight11 : 11,
    PivotTableStyleLight12 : 12,
    PivotTableStyleLight13 : 13,
    PivotTableStyleLight14 : 14,
    PivotTableStyleLight15 : 15,
    PivotTableStyleLight16 : 16,
    PivotTableStyleLight17 : 17,
    PivotTableStyleLight18 : 18,
    PivotTableStyleLight19 : 19,
    PivotTableStyleLight20 : 20,
    PivotTableStyleLight21 : 21,
    PivotTableStyleLight22 : 22,
    PivotTableStyleLight23 : 23,
    PivotTableStyleLight24 : 24,
    PivotTableStyleLight25 : 25,
    PivotTableStyleLight26 : 26,
    PivotTableStyleLight27 : 27,
    PivotTableStyleLight28 : 28,
    PivotTableStyleMedium1 : 29,
    PivotTableStyleMedium2 : 30,
    PivotTableStyleMedium3 : 31,
    PivotTableStyleMedium4 : 32,
    PivotTableStyleMedium5 : 33,
    PivotTableStyleMedium6 : 34,
    PivotTableStyleMedium7 : 35,
    PivotTableStyleMedium8 : 36,
    PivotTableStyleMedium9 : 37,
    PivotTableStyleMedium10 : 38,
    PivotTableStyleMedium11 : 39,
    PivotTableStyleMedium12 : 40,
    PivotTableStyleMedium13 : 41,
    PivotTableStyleMedium14 : 42,
    PivotTableStyleMedium15 : 43,
    PivotTableStyleMedium16 : 44,
    PivotTableStyleMedium17 : 45,
    PivotTableStyleMedium18 : 46,
    PivotTableStyleMedium19 : 47,
    PivotTableStyleMedium20 : 48,
    PivotTableStyleMedium21 : 49,
    PivotTableStyleMedium22 : 50,
    PivotTableStyleMedium23 : 51,
    PivotTableStyleMedium24 : 52,
    PivotTableStyleMedium25 : 53,
    PivotTableStyleMedium26 : 54,
    PivotTableStyleMedium27 : 55,
    PivotTableStyleMedium28 : 56,
    PivotTableStyleDark1 : 57,
    PivotTableStyleDark2 : 58,
    PivotTableStyleDark3 : 59,
    PivotTableStyleDark4 : 60,
    PivotTableStyleDark5 : 61,
    PivotTableStyleDark6 : 62,
    PivotTableStyleDark7 : 63,
    PivotTableStyleDark8 : 64,
    PivotTableStyleDark9 : 65,
    PivotTableStyleDark10 : 66,
    PivotTableStyleDark11 : 67,
    PivotTableStyleDark12 : 68,
    PivotTableStyleDark13 : 69,
    PivotTableStyleDark14 : 70,
    PivotTableStyleDark15 : 71,
    PivotTableStyleDark16 : 72,
    PivotTableStyleDark17 : 73,
    PivotTableStyleDark18 : 74,
    PivotTableStyleDark19 : 75,
    PivotTableStyleDark20 : 76,
    PivotTableStyleDark21 : 77,
    PivotTableStyleDark22 : 78,
    PivotTableStyleDark23 : 79,
    PivotTableStyleDark24 : 80,
    PivotTableStyleDark25 : 81,
    PivotTableStyleDark26 : 82,
    PivotTableStyleDark27 : 83,
    PivotTableStyleDark28 : 84,
    Custom : 85,
};

exports.DateTimeGroupingType = {
    Day : 0,
    Hour : 1,
    Minute : 2,
    Month : 3,
    Second : 4,
    Year : 5,
};

exports.DynamicFilterType = {
    AboveAverage : 0,
    BelowAverage : 1,
    LastMonth : 2,
    LastQuarter : 3,
    LastWeek : 4,
    LastYear : 5,
    January : 6,
    October : 7,
    November : 8,
    December : 9,
    February : 10,
    March : 11,
    April : 12,
    May : 13,
    June : 14,
    July : 15,
    August : 16,
    September : 17,
    NextMonth : 18,
    NextQuarter : 19,
    NextWeek : 20,
    NextYear : 21,
    None : 22,
    Quarter1 : 23,
    Quarter2 : 24,
    Quarter3 : 25,
    Quarter4 : 26,
    ThisMonth : 27,
    ThisQuarter : 28,
    ThisWeek : 29,
    ThisYear : 30,
    Today : 31,
    Tomorrow : 32,
    YearToDate : 33,
    Yesterday : 34,
};

exports.FilterType = {
    ColorFilter : 0,
    CustomFilters : 1,
    DynamicFilter : 2,
    MultipleFilters : 3,
    IconFilter : 4,
    Top10 : 5,
    None : 6,
};

exports.PdfCompliance = {
    None : 0,
    Pdf14 : 0,
    Pdf15 : 3,
    Pdf16 : 4,
    Pdf17 : 5,
    PdfA1b : 1,
    PdfA1a : 2,
    PdfA2b : 6,
    PdfA2u : 7,
    PdfA2a : 8,
    PdfA3b : 9,
    PdfA3u : 10,
    PdfA3a : 11,
};

exports.DirectoryType = {
    Volume : 0,
    SameVolume : 1,
    DownDirectory : 2,
    UpDirectory : 3,
};

exports.LightRigType = {
    Balanced : 0,
    BrightRoom : 1,
    Chilly : 2,
    Contrasting : 3,
    Flat : 4,
    Flood : 5,
    Freezing : 6,
    Glow : 7,
    Harsh : 8,
    LegacyFlat1 : 9,
    LegacyFlat2 : 10,
    LegacyFlat3 : 11,
    LegacyFlat4 : 12,
    LegacyHarsh1 : 13,
    LegacyHarsh2 : 14,
    LegacyHarsh3 : 15,
    LegacyHarsh4 : 16,
    LegacyNormal1 : 17,
    LegacyNormal2 : 18,
    LegacyNormal3 : 19,
    LegacyNormal4 : 20,
    Morning : 21,
    Soft : 22,
    Sunrise : 23,
    Sunset : 24,
    ThreePoint : 25,
    TwoPoint : 26,
    None : 27,
};

exports.SmartTagShowType = {
    All : 0,
    NoSmartTagIndicator : 1,
    None : 2,
};

exports.BevelType = {
    None : 0,
    Angle : 1,
    ArtDeco : 2,
    Circle : 3,
    Convex : 4,
    CoolSlant : 5,
    Cross : 6,
    Divot : 7,
    HardEdge : 8,
    RelaxedInset : 9,
    Riblet : 10,
    Slope : 11,
    SoftRound : 12,
};

exports.PresetCameraType = {
    IsometricBottomDown : 0,
    IsometricBottomUp : 1,
    IsometricLeftDown : 2,
    IsometricLeftUp : 3,
    IsometricOffAxis1Left : 4,
    IsometricOffAxis1Right : 5,
    IsometricOffAxis1Top : 6,
    IsometricOffAxis2Left : 7,
    IsometricOffAxis2Right : 8,
    IsometricOffAxis2Top : 9,
    IsometricOffAxis3Bottom : 10,
    IsometricOffAxis3Left : 11,
    IsometricOffAxis3Right : 12,
    IsometricOffAxis4Bottom : 13,
    IsometricOffAxis4Left : 14,
    IsometricOffAxis4Right : 15,
    IsometricRightDown : 16,
    IsometricRightUp : 17,
    IsometricTopDown : 18,
    IsometricTopUp : 19,
    LegacyObliqueBottom : 20,
    LegacyObliqueBottomLeft : 21,
    LegacyObliqueBottomRight : 22,
    LegacyObliqueFront : 23,
    LegacyObliqueLeft : 24,
    LegacyObliqueRight : 25,
    LegacyObliqueTop : 26,
    LegacyObliqueTopLeft : 27,
    LegacyObliqueTopRight : 28,
    LegacyPerspectiveBottom : 29,
    LegacyPerspectiveBottomLeft : 30,
    LegacyPerspectiveBottomRight : 31,
    LegacyPerspectiveFront : 32,
    LegacyPerspectiveLeft : 33,
    LegacyPerspectiveRight : 34,
    LegacyPerspectiveTop : 35,
    LegacyPerspectiveTopLeft : 36,
    LegacyPerspectiveTopRight : 37,
    ObliqueBottom : 38,
    ObliqueBottomLeft : 39,
    ObliqueBottomRight : 40,
    ObliqueLeft : 41,
    ObliqueRight : 42,
    ObliqueTop : 43,
    ObliqueTopLeft : 44,
    ObliqueTopRight : 45,
    OrthographicFront : 46,
    PerspectiveAbove : 47,
    PerspectiveAboveLeftFacing : 48,
    PerspectiveAboveRightFacing : 49,
    PerspectiveBelow : 50,
    PerspectiveContrastingLeftFacing : 51,
    PerspectiveContrastingRightFacing : 52,
    PerspectiveFront : 53,
    PerspectiveHeroicExtremeLeftFacing : 54,
    PerspectiveHeroicExtremeRightFacing : 55,
    PerspectiveHeroicLeftFacing : 56,
    PerspectiveHeroicRightFacing : 57,
    PerspectiveLeft : 58,
    PerspectiveRelaxed : 59,
    PerspectiveRelaxedModerately : 60,
    PerspectiveRight : 61,
};

exports.PresetShadowType = {
    NoShadow : 0,
    Custom : 1,
    OffsetDiagonalBottomRight : 2,
    OffsetBottom : 3,
    OffsetDiagonalBottomLeft : 4,
    OffsetRight : 5,
    OffsetCenter : 6,
    OffsetLeft : 7,
    OffsetDiagonalTopRight : 8,
    OffsetTop : 9,
    OffsetDiagonalTopLeft : 10,
    InsideDiagonalTopLeft : 11,
    InsideTop : 12,
    InsideDiagonalTopRight : 13,
    InsideLeft : 14,
    InsideCenter : 15,
    InsideRight : 16,
    InsideDiagonalBottomLeft : 17,
    InsideBottom : 18,
    InsideDiagonalBottomRight : 19,
    PerspectiveDiagonalUpperLeft : 20,
    PerspectiveDiagonalUpperRight : 21,
    Below : 22,
    PerspectiveDiagonalLowerLeft : 23,
    PerspectiveDiagonalLowerRight : 24,
};

exports.PresetMaterialType = {
    Clear : 0,
    DarkEdge : 1,
    Flat : 2,
    LegacyMatte : 3,
    LegacyMetal : 4,
    LegacyPlastic : 5,
    LegacyWireframe : 6,
    Matte : 7,
    Metal : 8,
    Plastic : 9,
    Powder : 10,
    SoftEdge : 11,
    SoftMetal : 12,
    TranslucentPowder : 13,
    WarmMatte : 14,
};

exports.RectangleAlignmentType = {
    Bottom : 0,
    BottomLeft : 1,
    BottomRight : 2,
    Center : 3,
    Left : 4,
    Right : 5,
    Top : 6,
    TopLeft : 7,
    TopRight : 8,
};

exports.FontSchemeType = {
    None : 0,
    Major : 1,
    Minor : 2,
};

exports.SaveFormat = {
    Csv : 1,
    CSV : 1,
    Xlsx : 6,
    Xlsm : 7,
    Xltx : 8,
    Xltm : 9,
    Xlam : 10,
    Tsv : 11,
    TSV : 11,
    TabDelimited : 11,
    Html : 12,
    MHtml : 17,
    Ods : 14,
    ODS : 14,
    Excel97To2003 : 5,
    SpreadsheetML : 15,
    Xlsb : 16,
    Auto : 0,
    Unknown : 255,
    Pdf : 13,
    Xps : 20,
    XPS : 20,
    Tiff : 21,
    TIFF : 21,
    Svg : 28,
    SVG : 28,
    Dif : 30,
    Ots : 31,
    Xlt : 32,
    Xml : 51,
    Numbers : 56,
    Markdown : 57,
    Fods : 59,
    FODS : 59,
    Sxc : 60,
    SXC : 60,
    Pptx : 61,
    Docx : 62,
    Emf : 258,
    Jpg : 261,
    Png : 262,
    Bmp : 263,
    Gif : 322,
    Json : 513,
    SqlScript : 514,
    XHtml : 771,
    Epub : 772,
    Azw3 : 773,
};

exports.SparklineAxisMinMaxType = {
    AutoIndividual : 0,
    Group : 1,
    Custom : 2,
};

exports.SparklinePresetStyleType = {
    Style1 : 0,
    Style2 : 1,
    Style3 : 2,
    Style4 : 3,
    Style5 : 4,
    Style6 : 5,
    Style7 : 6,
    Style8 : 7,
    Style9 : 8,
    Style10 : 9,
    Style11 : 10,
    Style12 : 11,
    Style13 : 12,
    Style14 : 13,
    Style15 : 14,
    Style16 : 15,
    Style17 : 16,
    Style18 : 17,
    Style19 : 18,
    Style20 : 19,
    Style21 : 20,
    Style22 : 21,
    Style23 : 22,
    Style24 : 23,
    Style25 : 24,
    Style26 : 25,
    Style27 : 26,
    Style28 : 27,
    Style29 : 28,
    Style30 : 29,
    Style31 : 30,
    Style32 : 31,
    Style33 : 32,
    Style34 : 33,
    Style35 : 34,
    Style36 : 35,
    Custom : 36,
};

exports.SparklineType = {
    Line : 0,
    Column : 1,
    Stacked : 2,
};

exports.BuiltinStyleType = {
    TwentyPercentAccent1 : 30,
    TwentyPercentAccent2 : 34,
    TwentyPercentAccent3 : 38,
    TwentyPercentAccent4 : 42,
    TwentyPercentAccent5 : 46,
    TwentyPercentAccent6 : 50,
    FortyPercentAccent1 : 31,
    FortyPercentAccent2 : 35,
    FortyPercentAccent3 : 39,
    FortyPercentAccent4 : 43,
    FortyPercentAccent5 : 47,
    FortyPercentAccent6 : 51,
    SixtyPercentAccent1 : 32,
    SixtyPercentAccent2 : 36,
    SixtyPercentAccent3 : 40,
    SixtyPercentAccent4 : 44,
    SixtyPercentAccent5 : 48,
    SixtyPercentAccent6 : 52,
    Accent1 : 29,
    Accent2 : 33,
    Accent3 : 37,
    Accent4 : 41,
    Accent5 : 45,
    Accent6 : 49,
    Bad : 27,
    Calculation : 22,
    CheckCell : 23,
    Comma : 3,
    Comma1 : 6,
    Currency : 4,
    Currency1 : 7,
    ExplanatoryText : 53,
    Good : 26,
    Header1 : 16,
    Header2 : 17,
    Header3 : 18,
    Header4 : 19,
    Hyperlink : 8,
    FollowedHyperlink : 9,
    Input : 20,
    LinkedCell : 24,
    Neutral : 28,
    Normal : 0,
    Note : 10,
    Output : 21,
    Percent : 5,
    Title : 15,
    Total : 25,
    WarningText : 11,
    RowLevel : 1,
    ColumnLevel : 2,
};

exports.AxisType = {
    Category : 0,
    Value : 1,
    Series : 2,
};

exports.CategoryType = {
    AutomaticScale : 0,
    CategoryScale : 1,
    TimeScale : 2,
};

exports.CrossType = {
    Automatic : 0,
    Maximum : 1,
    Minimum : 2,
    Custom : 3,
};

exports.TimeUnit = {
    Days : 0,
    Months : 1,
    Years : 2,
};

exports.ColorType = {
    Automatic : 0,
    AutomaticIndex : 1,
    RGB : 2,
    IndexedColor : 3,
    Theme : 4,
};

exports.LookAtType = {
    Contains : 0,
    StartWith : 1,
    EndWith : 2,
    EntireContent : 3,
};

exports.LookInType = {
    Formulas : 0,
    Values : 1,
    ValuesExcludeFormulaCell : 2,
    Comments : 3,
    OnlyFormulas : 4,
    OriginalValues : 5,
};

exports.AutoShapeType = {
    NotPrimitive : 0,
    Rectangle : 1,
    RoundedRectangle : 2,
    Oval : 3,
    Diamond : 4,
    IsoscelesTriangle : 5,
    RightTriangle : 6,
    Parallelogram : 7,
    Trapezoid : 8,
    Hexagon : 9,
    Octagon : 10,
    Cross : 11,
    Star5 : 12,
    RightArrow : 13,
    HomePlate : 15,
    Cube : 16,
    Balloon : 17,
    Seal : 18,
    Arc : 19,
    Line : 20,
    Plaque : 21,
    Can : 22,
    Donut : 23,
    TextSimple : 24,
    TextOctagon : 25,
    TextHexagon : 26,
    TextCurve : 27,
    TextWave : 28,
    TextRing : 29,
    TextOnCurve : 30,
    msosptTextOnRing : 31,
    StraightConnector : 32,
    BentConnector2 : 33,
    ElbowConnector : 34,
    BentConnector4 : 35,
    BentConnector5 : 36,
    CurvedConnector2 : 37,
    CurvedConnector : 38,
    CurvedConnector4 : 39,
    CurvedConnector5 : 40,
    LineCalloutNoBorder2 : 41,
    LineCalloutNoBorder3 : 42,
    LineCalloutNoBorder4 : 43,
    LineCalloutWithAccentBar2 : 44,
    LineCalloutWithAccentBar3 : 45,
    LineCalloutWithAccentBar4 : 46,
    LineCalloutWithBorder2 : 47,
    LineCalloutWithBorder3 : 48,
    LineCalloutWithBorder4 : 49,
    LineCalloutWithBorderAndAccentBar2 : 50,
    LineCalloutWithBorderAndAccentBar3 : 51,
    LineCalloutWithBorderAndAccentBar4 : 52,
    DownRibbon : 53,
    UpRibbon : 54,
    Chevron : 55,
    RegularPentagon : 56,
    NoSymbol : 57,
    Star8 : 58,
    Star16 : 59,
    Star32 : 60,
    RectangularCallout : 61,
    RoundedRectangularCallout : 62,
    OvalCallout : 63,
    Wave : 64,
    FoldedCorner : 65,
    LeftArrow : 66,
    DownArrow : 67,
    UpArrow : 68,
    LeftRightArrow : 69,
    UpDownArrow : 70,
    Explosion1 : 71,
    Explosion2 : 72,
    LightningBolt : 73,
    Heart : 74,
    PictureFrame : 75,
    QuadArrow : 76,
    LeftArrowCallout : 77,
    RightArrowCallout : 78,
    UpArrowCallout : 79,
    DownArrowCallout : 80,
    LeftRightArrowCallout : 81,
    UpDownArrowCallout : 82,
    QuadArrowCallout : 83,
    Bevel : 84,
    LeftBracket : 85,
    RightBracket : 86,
    LeftBrace : 87,
    RightBrace : 88,
    LeftUpArrow : 89,
    BentUpArrow : 90,
    BentArrow : 91,
    Star24 : 92,
    StripedRightArrow : 93,
    NotchedRightArrow : 94,
    BlockArc : 95,
    SmileyFace : 96,
    VerticalScroll : 97,
    HorizontalScroll : 98,
    CircularArrow : 99,
    NotchedCircularArrow : 100,
    UTurnArrow : 101,
    CurvedRightArrow : 102,
    CurvedLeftArrow : 103,
    CurvedUpArrow : 104,
    CurvedDownArrow : 105,
    CloudCallout : 106,
    CurvedDownRibbon : 107,
    CurvedUpRibbon : 108,
    FlowChartProcess : 109,
    FlowChartDecision : 110,
    FlowChartData : 111,
    FlowChartPredefinedProcess : 112,
    FlowChartInternalStorage : 113,
    FlowChartDocument : 114,
    FlowChartMultidocument : 115,
    FlowChartTerminator : 116,
    FlowChartPreparation : 117,
    FlowChartManualInput : 118,
    FlowChartManualOperation : 119,
    FlowChartConnector : 120,
    FlowChartCard : 121,
    FlowChartPunchedTape : 122,
    FlowChartSummingJunction : 123,
    FlowChartOr : 124,
    FlowChartCollate : 125,
    FlowChartSort : 126,
    FlowChartExtract : 127,
    FlowChartMerge : 128,
    FlowChartOfflineStorage : 129,
    FlowChartStoredData : 130,
    FlowChartSequentialAccessStorage : 131,
    FlowChartMagneticDisk : 132,
    FlowChartDirectAccessStorage : 133,
    FlowChartDisplay : 134,
    FlowChartDelay : 135,
    TextPlainText : 136,
    TextStop : 137,
    TextTriangle : 138,
    TextTriangleInverted : 139,
    TextChevron : 140,
    TextChevronInverted : 141,
    TextRingInside : 142,
    TextRingOutside : 143,
    TextArchUpCurve : 144,
    TextArchDownCurve : 145,
    TextCircleCurve : 146,
    TextButtonCurve : 147,
    TextArchUpPour : 148,
    TextArchDownPour : 149,
    TextCirclePour : 150,
    TextButtonPour : 151,
    TextCurveUp : 152,
    TextCurveDown : 153,
    TextCascadeUp : 154,
    TextCascadeDown : 155,
    TextWave1 : 156,
    TextWave2 : 157,
    TextDoubleWave1 : 158,
    TextDoubleWave2 : 159,
    TextInflate : 160,
    TextDeflate : 161,
    TextInflateBottom : 162,
    TextDeflateBottom : 163,
    TextInflateTop : 164,
    TextDeflateTop : 165,
    TextDeflateInflate : 166,
    TextDeflateInflateDeflate : 167,
    TextFadeRight : 168,
    TextFadeLeft : 169,
    TextFadeUp : 170,
    TextFadeDown : 171,
    TextSlantUp : 172,
    TextSlantDown : 173,
    TextCanUp : 174,
    TextCanDown : 175,
    FlowChartAlternateProcess : 176,
    FlowChartOffpageConnector : 177,
    LineCalloutNoBorder1 : 178,
    LineCalloutWithAccentBar1 : 179,
    LineCalloutWithBorder1 : 180,
    LineCalloutWithBorderAndAccentBar1 : 181,
    LeftRightUpArrow : 182,
    Sun : 183,
    Moon : 184,
    DoubleBracket : 185,
    DoubleBrace : 186,
    Star4 : 187,
    DoubleWave : 188,
    BlankActionButton : 189,
    HomeActionButton : 190,
    HelpActionButton : 191,
    InformationActionButton : 192,
    ForwardNextActionButton : 193,
    BackPreviousActionButton : 194,
    EndActionButton : 195,
    BeginningActionButton : 196,
    ReturnActionButton : 197,
    DocumentActionButton : 198,
    SoundActionButton : 199,
    MovieActionButton : 200,
    HostControl : 201,
    TextBox : 202,
    Heptagon : 2049,
    Decagon : 2050,
    Dodecagon : 2051,
    Star6 : 2052,
    Star7 : 2053,
    Star10 : 2054,
    Star12 : 2055,
    RoundSingleCornerRectangle : 2056,
    RoundSameSideCornerRectangle : 2057,
    RoundDiagonalCornerRectangle : 2058,
    SnipRoundSingleCornerRectangle : 2059,
    SnipSingleCornerRectangle : 2060,
    SnipSameSideCornerRectangle : 2061,
    SnipDiagonalCornerRectangle : 2062,
    Teardrop : 2063,
    Pie : 2064,
    HalfFrame : 2066,
    L_Shape : 2067,
    DiagonalStripe : 2068,
    Chord : 2069,
    Cloud : 2070,
    MathPlus : 2071,
    MathMinus : 2072,
    MathMultiply : 2073,
    MathDivide : 2074,
    MathEqual : 2075,
    MathNotEqual : 2076,
    LineInv : 2077,
    NonIsoscelesTrapezoid : 2078,
    PieWedge : 2079,
    LeftCircularArrow : 2080,
    LeftRightCircularArrow : 2081,
    SwooshArrow : 2082,
    LeftRightRibbon : 2083,
    TextNoShape : 2084,
    Gear6 : 2097,
    Gear9 : 2098,
    Funnel : 2099,
    CornerTabs : 2100,
    SquareTabs : 2101,
    PlaqueTabs : 2102,
    ChartX : 2103,
    ChartStar : 2104,
    ChartPlus : 2105,
    Frame : 2106,
    Model3D : 2107,
    RoundCallout : 2108,
    TextArchLeftPour : 2109,
    TextArchRightPour : 2110,
    TextArchLeftCurve : 2111,
    TextArchRightCurve : 2112,
    Unknown : 4095,
};

exports.BackgroundMode = {
    Automatic : 0,
    Opaque : 1,
    Transparent : 2,
};

exports.BackgroundType = {
    DiagonalCrosshatch : 9,
    DiagonalStripe : 8,
    Gray6 : 18,
    Gray12 : 17,
    Gray25 : 4,
    Gray50 : 2,
    Gray75 : 3,
    HorizontalStripe : 5,
    None : 0,
    ReverseDiagonalStripe : 7,
    Solid : 1,
    ThickDiagonalCrosshatch : 10,
    ThinDiagonalCrosshatch : 16,
    ThinDiagonalStripe : 14,
    ThinHorizontalCrosshatch : 15,
    ThinHorizontalStripe : 11,
    ThinReverseDiagonalStripe : 13,
    ThinVerticalStripe : 12,
    VerticalStripe : 6,
};

exports.Bar3DShapeType = {
    Box : 0,
    PyramidToPoint : 1,
    PyramidToMax : 2,
    Cylinder : 3,
    ConeToPoint : 4,
    ConeToMax : 5,
};

exports.BorderType = {
    BottomBorder : 8,
    DiagonalDown : 16,
    DiagonalUp : 32,
    LeftBorder : 1,
    RightBorder : 2,
    TopBorder : 4,
    Horizontal : 128,
    Vertical : 64,
};

exports.BubbleSizeRepresents = {
    SizeIsArea : 0,
    SizeIsWidth : 1,
};

exports.CalcModeType = {
    Automatic : 0,
    AutomaticExceptTable : 1,
    Manual : 2,
};

exports.CellBorderType = {
    DashDot : 9,
    DashDotDot : 11,
    Dashed : 3,
    Dotted : 4,
    Double : 6,
    Hair : 7,
    MediumDashDot : 10,
    MediumDashDotDot : 12,
    MediumDashed : 8,
    None : 0,
    Medium : 2,
    SlantedDashDot : 13,
    Thick : 5,
    Thin : 1,
};

exports.CellValueType = {
    IsUnknown : 0,
    IsNull : 1,
    IsNumeric : 2,
    IsDateTime : 4,
    IsString : 8,
    IsBool : 16,
    IsError : 32,
};

exports.ChartMarkerType = {
    Automatic : 0,
    Circle : 1,
    Dash : 2,
    Diamond : 3,
    Dot : 4,
    None : 5,
    SquarePlus : 6,
    Square : 7,
    SquareStar : 8,
    Triangle : 9,
    SquareX : 10,
    Picture : 11,
};

exports.ChartSplitType = {
    Position : 0,
    Value : 1,
    PercentValue : 2,
    Custom : 3,
    Auto : 4,
};

exports.ChartType = {
    Area : 0,
    AreaStacked : 1,
    Area100PercentStacked : 2,
    Area3D : 3,
    Area3DStacked : 4,
    Area3D100PercentStacked : 5,
    Bar : 6,
    BarStacked : 7,
    Bar100PercentStacked : 8,
    Bar3DClustered : 9,
    Bar3DStacked : 10,
    Bar3D100PercentStacked : 11,
    Bubble : 12,
    Bubble3D : 13,
    Column : 14,
    ColumnStacked : 15,
    Column100PercentStacked : 16,
    Column3D : 17,
    Column3DClustered : 18,
    Column3DStacked : 19,
    Column3D100PercentStacked : 20,
    Cone : 21,
    ConeStacked : 22,
    Cone100PercentStacked : 23,
    ConicalBar : 24,
    ConicalBarStacked : 25,
    ConicalBar100PercentStacked : 26,
    ConicalColumn3D : 27,
    Cylinder : 28,
    CylinderStacked : 29,
    Cylinder100PercentStacked : 30,
    CylindricalBar : 31,
    CylindricalBarStacked : 32,
    CylindricalBar100PercentStacked : 33,
    CylindricalColumn3D : 34,
    Doughnut : 35,
    DoughnutExploded : 36,
    Line : 37,
    LineStacked : 38,
    Line100PercentStacked : 39,
    LineWithDataMarkers : 40,
    LineStackedWithDataMarkers : 41,
    Line100PercentStackedWithDataMarkers : 42,
    Line3D : 43,
    Pie : 44,
    Pie3D : 45,
    PiePie : 46,
    PieExploded : 47,
    Pie3DExploded : 48,
    PieBar : 49,
    Pyramid : 50,
    PyramidStacked : 51,
    Pyramid100PercentStacked : 52,
    PyramidBar : 53,
    PyramidBarStacked : 54,
    PyramidBar100PercentStacked : 55,
    PyramidColumn3D : 56,
    Radar : 57,
    RadarWithDataMarkers : 58,
    RadarFilled : 59,
    Scatter : 60,
    ScatterConnectedByCurvesWithDataMarker : 61,
    ScatterConnectedByCurvesWithoutDataMarker : 62,
    ScatterConnectedByLinesWithDataMarker : 63,
    ScatterConnectedByLinesWithoutDataMarker : 64,
    StockHighLowClose : 65,
    StockOpenHighLowClose : 66,
    StockVolumeHighLowClose : 67,
    StockVolumeOpenHighLowClose : 68,
    Surface3D : 69,
    SurfaceWireframe3D : 70,
    SurfaceContour : 71,
    SurfaceContourWireframe : 72,
    BoxWhisker : 73,
    Funnel : 74,
    ParetoLine : 75,
    Sunburst : 76,
    Treemap : 77,
    Waterfall : 78,
    Histogram : 79,
    Map : 80,
    RadialHistogram : 81,
};

exports.CheckValueType = {
    UnChecked : 0,
    Checked : 1,
    Mixed : 2,
};

exports.ConsolidationFunction = {
    Sum : 0,
    Count : 1,
    Average : 2,
    Max : 3,
    Min : 4,
    Product : 5,
    CountNums : 6,
    StdDev : 7,
    StdDevp : 8,
    Var : 9,
    Varp : 10,
    DistinctCount : 11,
};

exports.ImageType = {
    Unknown : 0,
    Emf : 2,
    Wmf : 3,
    Pict : 4,
    Jpeg : 5,
    Png : 6,
    Bmp : 7,
    Gif : 66,
    Tiff : 67,
    Svg : 68,
    Svm : 69,
    Gltf : 70,
    OfficeCompatibleEmf : 71,
    WebP : 72,
};

exports.PdfCompressionCore = {
    None : 0,
    Rle : 1,
    Lzw : 2,
    Flate : 3,
};

exports.CountryCode = {
    Default : 0,
    USA : 1,
    Canada : 2,
    LatinAmeric : 3,
    Russia : 7,
    Egypt : 20,
    Greece : 30,
    Netherlands : 31,
    Belgium : 32,
    France : 33,
    Spain : 34,
    Hungary : 36,
    Italy : 39,
    Switzerland : 41,
    Austria : 43,
    UnitedKingdom : 44,
    Denmark : 45,
    Sweden : 46,
    Norway : 47,
    Poland : 48,
    Germany : 49,
    Mexico : 52,
    Brazil : 55,
    Australia : 61,
    NewZealand : 64,
    Thailand : 66,
    Japan : 81,
    SouthKorea : 82,
    VietNam : 84,
    China : 86,
    Turkey : 90,
    India : 91,
    Algeria : 213,
    Morocco : 216,
    Libya : 218,
    Portugal : 351,
    Iceland : 354,
    Finland : 358,
    Czech : 420,
    Taiwan : 886,
    Lebanon : 961,
    Jordan : 962,
    Syria : 963,
    Iraq : 964,
    Kuwait : 965,
    Saudi : 966,
    UnitedArabEmirates : 971,
    Israel : 972,
    Qatar : 974,
    Iran : 981,
};

exports.DataLabelsSeparatorType = {
    Auto : 0,
    Space : 1,
    Comma : 2,
    Semicolon : 3,
    Period : 4,
    NewLine : 5,
    Custom : 6,
};

exports.DisplayDrawingObjects = {
    DisplayShapes : 0,
    Placeholders : 1,
    Hide : 2,
};

exports.DisplayUnitType = {
    None : 0,
    Hundreds : 1,
    Thousands : 2,
    TenThousands : 3,
    HundredThousands : 4,
    Millions : 5,
    TenMillions : 6,
    HundredMillions : 7,
    Billions : 8,
    Trillions : 9,
    Percentage : 10,
    Cust : 11,
    Custom : 12,
};

exports.PropertyType = {
    Boolean : 0,
    DateTime : 1,
    Double : 2,
    Number : 3,
    String : 4,
    Blob : 5,
};

exports.EncryptionType = {
    XOR : 0,
    Compatible : 1,
    EnhancedCryptographicProviderV1 : 2,
    StrongCryptographicProvider : 3,
};

exports.ErrorBarType = {
    Custom : 0,
    FixedValue : 1,
    Percent : 2,
    StDev : 3,
    StError : 4,
};

exports.ExceptionType = {
    Chart : 0,
    DataType : 1,
    DataValidation : 2,
    ConditionalFormatting : 3,
    FileFormat : 4,
    Formula : 5,
    InvalidData : 6,
    InvalidOperator : 7,
    IncorrectPassword : 8,
    License : 9,
    Limitation : 10,
    PageSetup : 11,
    PivotTable : 12,
    Shape : 13,
    Sparkline : 14,
    SheetName : 15,
    SheetType : 16,
    Interrupted : 17,
    IO : 18,
    Permission : 19,
    UnsupportedFeature : 20,
    UnsupportedStream : 21,
    UndisclosedInformation : 22,
    FileCorrupted : 23,
    Internal : 24,
};

exports.FileFormatType = {
    Csv : 1,
    Xlsx : 6,
    Xlsm : 7,
    Xltx : 8,
    Xltm : 9,
    Xlam : 10,
    Tsv : 11,
    TabDelimited : 11,
    Html : 12,
    MHtml : 17,
    Ods : 14,
    Excel97To2003 : 5,
    SpreadsheetML : 15,
    Xlsb : 16,
    Unknown : 255,
    Pdf : 13,
    Xps : 20,
    Tiff : 21,
    Svg : 28,
    Excel95 : 22,
    Excel4 : 23,
    Excel3 : 24,
    Excel2 : 25,
    Pptx : 26,
    Docx : 27,
    Dif : 30,
    Doc : 31,
    Ppt : 32,
    MapiMessage : 33,
    MsEquation : 34,
    Ole10Native : 35,
    Vsd : 36,
    Vsdx : 37,
    Docm : 38,
    Dotx : 39,
    Dotm : 40,
    Pptm : 41,
    Potx : 42,
    Potm : 43,
    Ppsx : 44,
    Ppsm : 45,
    Ooxml : 46,
    Odt : 47,
    Odp : 48,
    Odf : 49,
    Odg : 50,
    Xml : 51,
    Xlt : 52,
    Ott : 53,
    Bmp : 54,
    Ots : 55,
    Numbers : 56,
    Numbers09 : 56,
    Markdown : 57,
    GraphChart : 58,
    Fods : 59,
    Sxc : 60,
    Otp : 61,
    Numbers35 : 62,
    Emf : 258,
    Wmf : 259,
    Jpg : 261,
    Png : 262,
    Gif : 322,
    WebP : 323,
    Json : 513,
    SqlScript : 514,
    XHtml : 771,
    OneNote : 772,
    MicrosoftCabinet : 773,
    Rtf : 774,
    Epub : 775,
    Azw3 : 777,
    Oxps : 776,
    GZip : 35615,
};

exports.FillPattern = {
    None : 0,
    Solid : 1,
    Gray5 : 2,
    Gray10 : 3,
    Gray20 : 4,
    Gray30 : 5,
    Gray40 : 6,
    Gray50 : 7,
    Gray60 : 8,
    Gray70 : 9,
    Gray75 : 10,
    Gray80 : 11,
    Gray90 : 12,
    Gray25 : 13,
    LightDownwardDiagonal : 14,
    LightUpwardDiagonal : 15,
    DarkDownwardDiagonal : 16,
    DarkUpwardDiagonal : 17,
    WideDownwardDiagonal : 18,
    WideUpwardDiagonal : 19,
    LightVertical : 20,
    LightHorizontal : 21,
    NarrowVertical : 22,
    NarrowHorizontal : 23,
    DarkVertical : 24,
    DarkHorizontal : 25,
    DashedDownwardDiagonal : 26,
    DashedUpwardDiagonal : 27,
    DashedVertical : 28,
    DashedHorizontal : 29,
    SmallConfetti : 30,
    LargeConfetti : 31,
    ZigZag : 32,
    Wave : 33,
    DiagonalBrick : 34,
    HorizontalBrick : 35,
    Weave : 36,
    Plaid : 37,
    Divot : 38,
    DottedGrid : 39,
    DottedDiamond : 40,
    Shingle : 41,
    Trellis : 42,
    Sphere : 43,
    SmallGrid : 44,
    LargeGrid : 45,
    SmallCheckerBoard : 46,
    LargeCheckerBoard : 47,
    OutlinedDiamond : 48,
    SolidDiamond : 49,
    Unknown : 50,
};

exports.FilterOperatorType = {
    LessOrEqual : 0,
    LessThan : 1,
    Equal : 2,
    GreaterThan : 3,
    NotEqual : 4,
    GreaterOrEqual : 5,
    None : 6,
    BeginsWith : 7,
    EndsWith : 8,
    Contains : 9,
    NotContains : 10,
};

exports.FontUnderlineType = {
    None : 0,
    Single : 1,
    Double : 2,
    Accounting : 3,
    DoubleAccounting : 4,
    Dash : 5,
    DashDotDotHeavy : 6,
    DashDotHeavy : 7,
    DashedHeavy : 8,
    DashLong : 9,
    DashLongHeavy : 10,
    DotDash : 11,
    DotDotDash : 12,
    Dotted : 13,
    DottedHeavy : 14,
    Heavy : 15,
    Wave : 16,
    WavyDouble : 17,
    WavyHeavy : 18,
    Words : 19,
};

exports.FormatConditionType = {
    CellValue : 1,
    Expression : 2,
    Top10 : 4,
    UniqueValues : 8,
    DuplicateValues : 16,
    ContainsText : 32,
    NotContainsText : 64,
    BeginsWith : 128,
    EndsWith : 256,
    ContainsBlanks : 512,
    NotContainsBlanks : 1024,
    ContainsErrors : 2048,
    NotContainsErrors : 4096,
    TimePeriod : 8192,
    AboveAverage : 16384,
    ColorScale : 32768,
    DataBar : 65536,
    IconSet : 131072,
};

exports.FormattingType = {
    Automatic : 0,
    Custom : 2,
    None : 1,
};

exports.GradientColorType = {
    None : 0,
    OneColor : 1,
    PresetColors : 2,
    TwoColors : 3,
};

exports.GradientPresetType = {
    Brass : 0,
    CalmWater : 1,
    Chrome : 2,
    ChromeII : 3,
    Daybreak : 4,
    Desert : 5,
    EarlySunset : 6,
    Fire : 7,
    Fog : 8,
    Gold : 9,
    GoldII : 10,
    Horizon : 11,
    LateSunset : 12,
    Mahogany : 13,
    Moss : 14,
    Nightfall : 15,
    Ocean : 16,
    Parchment : 17,
    Peacock : 18,
    Rainbow : 19,
    RainbowII : 20,
    Sapphire : 21,
    Silver : 22,
    Wheat : 23,
    Unknown : 24,
};

exports.GradientStyleType = {
    DiagonalDown : 0,
    DiagonalUp : 1,
    FromCenter : 2,
    FromCorner : 3,
    Horizontal : 4,
    Vertical : 5,
    Unknown : 6,
};

exports.LegendPositionType = {
    Bottom : 0,
    Corner : 1,
    Left : 4,
    NotDocked : 7,
    Right : 3,
    Top : 2,
};

exports.LineType = {
    DarkGray : 6,
    Dash : 1,
    DashDot : 3,
    DashDotDot : 4,
    Dot : 2,
    LightGray : 8,
    MediumGray : 7,
    Solid : 0,
};

exports.MsoArrowheadStyle = {
    None : 0,
    Arrow : 1,
    ArrowStealth : 2,
    ArrowDiamond : 3,
    ArrowOval : 4,
    ArrowOpen : 5,
};

exports.ChartLineFormattingType = {
    Automatic : 0,
    Solid : 1,
    None : 2,
    Gradient : 3,
};

exports.GradientFillType = {
    Linear : 0,
    Radial : 1,
    Rectangle : 2,
    Path : 3,
};

exports.GradientDirectionType = {
    FromUpperLeftCorner : 0,
    FromUpperRightCorner : 1,
    FromLowerLeftCorner : 2,
    FromLowerRightCorner : 3,
    FromCenter : 4,
    Unknown : 5,
};

exports.LineCapType = {
    Square : 0,
    Round : 1,
    Flat : 2,
    None : 3,
};

exports.LineJoinType = {
    Round : 0,
    Bevel : 1,
    Miter : 2,
    None : 3,
};

exports.PlotEmptyCellsType = {
    NotPlotted : 0,
    Zero : 1,
    Interpolated : 2,
};

exports.FillPictureType = {
    Stretch : 0,
    Stack : 1,
    StackAndScale : 2,
};

exports.MsoDrawingType = {
    Group : 0,
    Line : 1,
    Rectangle : 2,
    Oval : 3,
    Arc : 4,
    Chart : 5,
    TextBox : 6,
    Button : 7,
    Picture : 8,
    Polygon : 9,
    CheckBox : 11,
    RadioButton : 12,
    Label : 14,
    Dialog_Box : 15,
    Spinner : 16,
    ScrollBar : 17,
    ListBox : 18,
    GroupBox : 19,
    ComboBox : 20,
    Comment : 25,
    OleObject : 24,
    CellsDrawing : 30,
    Unknown : 29,
    Slicer : 31,
    WebExtension : 32,
    SmartArt : 33,
    CustomXml : 34,
    Timeline : 35,
    Model3D : 36,
};

exports.MsoPresetTextEffect = {
    TextEffect1 : 0,
    TextEffect2 : 1,
    TextEffect3 : 2,
    TextEffect4 : 3,
    TextEffect5 : 4,
    TextEffect6 : 5,
    TextEffect7 : 6,
    TextEffect8 : 7,
    TextEffect9 : 8,
    TextEffect10 : 9,
    TextEffect11 : 10,
    TextEffect12 : 11,
    TextEffect13 : 12,
    TextEffect14 : 13,
    TextEffect15 : 14,
    TextEffect16 : 15,
    TextEffect17 : 16,
    TextEffect18 : 17,
    TextEffect19 : 18,
    TextEffect20 : 19,
    TextEffect21 : 20,
    TextEffect22 : 21,
    TextEffect23 : 22,
    TextEffect24 : 23,
    TextEffect25 : 24,
    TextEffect26 : 25,
    TextEffect27 : 26,
    TextEffect28 : 27,
    TextEffect29 : 28,
    TextEffect30 : 29,
};

exports.MsoPresetTextEffectShape = {
    PlainText : 136,
    Stop : 137,
    TriangleUp : 138,
    TriangleDown : 139,
    ChevronUp : 140,
    ChevronDown : 141,
    RingInside : 142,
    RingOutside : 143,
    ArchUpCurve : 144,
    ArchDownCurve : 145,
    CircleCurve : 146,
    ButtonCurve : 147,
    ArchUpPour : 148,
    ArchDownPour : 149,
    CirclePour : 150,
    ButtonPour : 151,
    CurveUp : 152,
    CurveDown : 153,
    CanUp : 174,
    CanDown : 175,
    Wave1 : 156,
    Wave2 : 157,
    DoubleWave1 : 158,
    DoubleWave2 : 159,
    Inflate : 160,
    Deflate : 161,
    InflateBottom : 162,
    DeflateBottom : 163,
    InflateTop : 164,
    DeflateTop : 165,
    DeflateInflate : 166,
    DeflateInflateDeflate : 167,
    FadeRight : 168,
    FadeLeft : 169,
    FadeUp : 170,
    FadeDown : 171,
    SlantUp : 172,
    SlantDown : 173,
    CascadeUp : 154,
    CascadeDown : 155,
    Mixed : 255,
};

exports.MsoLineDashStyle = {
    Dash : 0,
    DashDot : 1,
    DashDotDot : 2,
    DashLongDash : 3,
    DashLongDashDot : 4,
    RoundDot : 5,
    Solid : 6,
    SquareDot : 7,
    Custom : 8,
};

exports.MsoLineStyle = {
    Single : 0,
    ThickBetweenThin : 1,
    ThinThick : 2,
    ThickThin : 3,
    ThinThin : 4,
};

exports.OperatorType = {
    Between : 0,
    Equal : 1,
    GreaterThan : 2,
    GreaterOrEqual : 3,
    LessThan : 4,
    LessOrEqual : 5,
    None : 6,
    NotBetween : 7,
    NotEqual : 8,
};

exports.PageOrientationType = {
    Landscape : 0,
    Portrait : 1,
};

exports.PaperSizeType = {
    PaperLetter : 1,
    PaperLetterSmall : 2,
    PaperTabloid : 3,
    PaperLedger : 4,
    PaperLegal : 5,
    PaperStatement : 6,
    PaperExecutive : 7,
    PaperA3 : 8,
    PaperA4 : 9,
    PaperA4Small : 10,
    PaperA5 : 11,
    PaperB4 : 12,
    PaperB5 : 13,
    PaperFolio : 14,
    PaperQuarto : 15,
    Paper10x14 : 16,
    Paper11x17 : 17,
    PaperNote : 18,
    PaperEnvelope9 : 19,
    PaperEnvelope10 : 20,
    PaperEnvelope11 : 21,
    PaperEnvelope12 : 22,
    PaperEnvelope14 : 23,
    PaperCSheet : 24,
    PaperDSheet : 25,
    PaperESheet : 26,
    PaperEnvelopeDL : 27,
    PaperEnvelopeC5 : 28,
    PaperEnvelopeC3 : 29,
    PaperEnvelopeC4 : 30,
    PaperEnvelopeC6 : 31,
    PaperEnvelopeC65 : 32,
    PaperEnvelopeB4 : 33,
    PaperEnvelopeB5 : 34,
    PaperEnvelopeB6 : 35,
    PaperEnvelopeItaly : 36,
    PaperEnvelopeMonarch : 37,
    PaperEnvelopePersonal : 38,
    PaperFanfoldUS : 39,
    PaperFanfoldStdGerman : 40,
    PaperFanfoldLegalGerman : 41,
    PaperISOB4 : 42,
    PaperJapanesePostcard : 43,
    Paper9x11 : 44,
    Paper10x11 : 45,
    Paper15x11 : 46,
    PaperEnvelopeInvite : 47,
    PaperLetterExtra : 50,
    PaperLegalExtra : 51,
    PaperTabloidExtra : 52,
    PaperA4Extra : 53,
    PaperLetterTransverse : 54,
    PaperA4Transverse : 55,
    PaperLetterExtraTransverse : 56,
    PaperSuperA : 57,
    PaperSuperB : 58,
    PaperLetterPlus : 59,
    PaperA4Plus : 60,
    PaperA5Transverse : 61,
    PaperJISB5Transverse : 62,
    PaperA3Extra : 63,
    PaperA5Extra : 64,
    PaperISOB5Extra : 65,
    PaperA2 : 66,
    PaperA3Transverse : 67,
    PaperA3ExtraTransverse : 68,
    PaperJapaneseDoublePostcard : 69,
    PaperA6 : 70,
    PaperJapaneseEnvelopeKaku2 : 71,
    PaperJapaneseEnvelopeKaku3 : 72,
    PaperJapaneseEnvelopeChou3 : 73,
    PaperJapaneseEnvelopeChou4 : 74,
    PaperLetterRotated : 75,
    PaperA3Rotated : 76,
    PaperA4Rotated : 77,
    PaperA5Rotated : 78,
    PaperJISB4Rotated : 79,
    PaperJISB5Rotated : 80,
    PaperJapanesePostcardRotated : 81,
    PaperJapaneseDoublePostcardRotated : 82,
    PaperA6Rotated : 83,
    PaperJapaneseEnvelopeKaku2Rotated : 84,
    PaperJapaneseEnvelopeKaku3Rotated : 85,
    PaperJapaneseEnvelopeChou3Rotated : 86,
    PaperJapaneseEnvelopeChou4Rotated : 87,
    PaperJISB6 : 88,
    PaperJISB6Rotated : 89,
    Paper12x11 : 90,
    PaperJapaneseEnvelopeYou4 : 91,
    PaperJapaneseEnvelopeYou4Rotated : 92,
    PaperPRC16K : 93,
    PaperPRC32K : 94,
    PaperPRCBig32K : 95,
    PaperPRCEnvelope1 : 96,
    PaperPRCEnvelope2 : 97,
    PaperPRCEnvelope3 : 98,
    PaperPRCEnvelope4 : 99,
    PaperPRCEnvelope5 : 100,
    PaperPRCEnvelope6 : 101,
    PaperPRCEnvelope7 : 102,
    PaperPRCEnvelope8 : 103,
    PaperPRCEnvelope9 : 104,
    PaperPRCEnvelope10 : 105,
    PaperPRC16KRotated : 106,
    PaperPRC32KRotated : 107,
    PaperPRCBig32KRotated : 108,
    PaperPRCEnvelope1Rotated : 109,
    PaperPRCEnvelope2Rotated : 110,
    PaperPRCEnvelope3Rotated : 111,
    PaperPRCEnvelope4Rotated : 112,
    PaperPRCEnvelope5Rotated : 113,
    PaperPRCEnvelope6Rotated : 114,
    PaperPRCEnvelope7Rotated : 115,
    PaperPRCEnvelope8Rotated : 116,
    PaperPRCEnvelope9Rotated : 117,
    PaperPRCEnvelope10Rotated : 118,
    PaperB3 : 300,
    PaperBusinessCard : 301,
    PaperThermal : 302,
    Custom : 0,
};

exports.PlacementType = {
    FreeFloating : 0,
    Move : 1,
    MoveAndSize : 2,
};

exports.PrintCommentsType = {
    PrintInPlace : 0,
    PrintNoComments : 1,
    PrintSheetEnd : 2,
};

exports.PrintErrorsType = {
    PrintErrorsBlank : 0,
    PrintErrorsDash : 1,
    PrintErrorsDisplayed : 2,
    PrintErrorsNA : 3,
};

exports.PrintOrderType = {
    DownThenOver : 0,
    OverThenDown : 1,
};

exports.PrintSizeType = {
    Full : 0,
    Fit : 1,
    Custom : 2,
};

exports.ProtectionType = {
    All : 0,
    Contents : 1,
    Objects : 2,
    Scenarios : 3,
    Structure : 4,
    Windows : 5,
    None : 6,
};

exports.SelectionType = {
    Single : 0,
    Multi : 1,
    Extend : 2,
};

exports.SheetType = {
    VB : 0,
    Worksheet : 1,
    Chart : 2,
    BIFF4Macro : 3,
    InternationalMacro : 4,
    Other : 5,
    Dialog : 6,
};

exports.ShiftType = {
    Down : 0,
    Left : 1,
    None : 2,
    Right : 3,
    Up : 4,
};

exports.SortOrder = {
    Ascending : 0,
    Descending : 1,
};

exports.StyleModifyFlag = {
    LeftBorder : 256,
    RightBorder : 512,
    TopBorder : 1024,
    BottomBorder : 2048,
    DiagonalDownBorder : 4096,
    DiagonalUpBorder : 8192,
    Diagonal : 12288,
    HorizontalBorder : 32,
    VerticalBorder : 64,
    Borders : 16224,
    NumberFormat : 16384,
    HorizontalAlignment : 32768,
    VerticalAlignment : 65536,
    Indent : 131072,
    Rotation : 262144,
    WrapText : 524288,
    ShrinkToFit : 1048576,
    TextDirection : 2097152,
    RelativeIndent : 128,
    AlignmentSettings : 4161664,
    Pattern : 4194304,
    ForegroundColor : 8388608,
    BackgroundColor : 16777216,
    CellShading : 29360128,
    Locked : 67108864,
    HideFormula : 134217728,
    ProtectionSettings : 201326592,
    FontSize : 1,
    FontName : 2,
    FontColor : 3,
    FontWeight : 4,
    FontItalic : 5,
    FontUnderline : 6,
    FontStrike : 7,
    FontScript : 8,
    FontFamily : 9,
    FontCharset : 10,
    FontScheme : 11,
    FontDirty : 12,
    FontSpellingError : 13,
    FontUFillTx : 14,
    FontSpacing : 15,
    FontKerning : 16,
    FontEqualize : 17,
    FontCap : 18,
    FontVerticalText : 19,
    Font : 31,
    All : 234881023,
};

exports.TableDataSourceType = {
    Worksheet : 0,
    SharePoint : 1,
    XML : 2,
    QueryTable : 3,
};

exports.TableStyleElementType = {
    BlankRow : 18,
    FirstColumn : 8,
    FirstColumnStripe : 3,
    FirstColumnSubheading : 22,
    FirstHeaderCell : 11,
    FirstRowStripe : 5,
    FirstRowSubheading : 25,
    FirstSubtotalColumn : 15,
    FirstSubtotalRow : 19,
    GrandTotalColumn : 28,
    GrandTotalRow : 29,
    FirstTotalCell : 13,
    HeaderRow : 9,
    LastColumn : 7,
    LastHeaderCell : 12,
    LastTotalCell : 14,
    PageFieldLabels : 1,
    PageFieldValues : 2,
    SecondColumnStripe : 4,
    SecondColumnSubheading : 23,
    SecondRowStripe : 6,
    SecondRowSubheading : 26,
    SecondSubtotalColumn : 16,
    SecondSubtotalRow : 20,
    ThirdColumnSubheading : 24,
    ThirdRowSubheading : 27,
    ThirdSubtotalColumn : 17,
    ThirdSubtotalRow : 21,
    TotalRow : 10,
    WholeTable : 0,
};

exports.TableStyleType = {
    None : 0,
    TableStyleLight1 : 1,
    TableStyleLight2 : 2,
    TableStyleLight3 : 3,
    TableStyleLight4 : 4,
    TableStyleLight5 : 5,
    TableStyleLight6 : 6,
    TableStyleLight7 : 7,
    TableStyleLight8 : 8,
    TableStyleLight9 : 9,
    TableStyleLight10 : 10,
    TableStyleLight11 : 11,
    TableStyleLight12 : 12,
    TableStyleLight13 : 13,
    TableStyleLight14 : 14,
    TableStyleLight15 : 15,
    TableStyleLight16 : 16,
    TableStyleLight17 : 17,
    TableStyleLight18 : 18,
    TableStyleLight19 : 19,
    TableStyleLight20 : 20,
    TableStyleLight21 : 21,
    TableStyleMedium1 : 22,
    TableStyleMedium2 : 23,
    TableStyleMedium3 : 24,
    TableStyleMedium4 : 25,
    TableStyleMedium5 : 26,
    TableStyleMedium6 : 27,
    TableStyleMedium7 : 28,
    TableStyleMedium8 : 29,
    TableStyleMedium9 : 30,
    TableStyleMedium10 : 31,
    TableStyleMedium11 : 32,
    TableStyleMedium12 : 33,
    TableStyleMedium13 : 34,
    TableStyleMedium14 : 35,
    TableStyleMedium15 : 36,
    TableStyleMedium16 : 37,
    TableStyleMedium17 : 38,
    TableStyleMedium18 : 39,
    TableStyleMedium19 : 40,
    TableStyleMedium20 : 41,
    TableStyleMedium21 : 42,
    TableStyleMedium22 : 43,
    TableStyleMedium23 : 44,
    TableStyleMedium24 : 45,
    TableStyleMedium25 : 46,
    TableStyleMedium26 : 47,
    TableStyleMedium27 : 48,
    TableStyleMedium28 : 49,
    TableStyleDark1 : 50,
    TableStyleDark2 : 51,
    TableStyleDark3 : 52,
    TableStyleDark4 : 53,
    TableStyleDark5 : 54,
    TableStyleDark6 : 55,
    TableStyleDark7 : 56,
    TableStyleDark8 : 57,
    TableStyleDark9 : 58,
    TableStyleDark10 : 59,
    TableStyleDark11 : 60,
    Custom : 61,
};

exports.TextAlignmentType = {
    General : 1792,
    Bottom : 528,
    Center : 1802,
    CenterAcross : 1286,
    Distributed : 807,
    Fill : 1284,
    Justify : 1821,
    Left : 257,
    Right : 259,
    Top : 512,
    JustifiedLow : 832,
    ThaiDistributed : 896,
};

exports.TextDirectionType = {
    Context : 0,
    LeftToRight : 1,
    RightToLeft : 2,
};

exports.TextOrientationType = {
    ClockWise : 0,
    CounterClockWise : 1,
    NoRotation : 2,
    TopToBottom : 3,
};

exports.TextureType = {
    BlueTissuePaper : 0,
    Bouquet : 1,
    BrownMarble : 2,
    Canvas : 3,
    Cork : 4,
    Denim : 5,
    FishFossil : 6,
    Granite : 7,
    GreenMarble : 8,
    MediumWood : 9,
    Newsprint : 10,
    Oak : 11,
    PaperBag : 12,
    Papyrus : 13,
    Parchment : 14,
    PinkTissuePaper : 15,
    PurpleMesh : 16,
    RecycledPaper : 17,
    Sand : 18,
    Stationery : 19,
    Walnut : 20,
    WaterDroplets : 21,
    WhiteMarble : 22,
    WovenMat : 23,
    Unknown : 24,
};

exports.ThemeColorType = {
    Background1 : 0,
    Text1 : 1,
    Background2 : 2,
    Text2 : 3,
    Accent1 : 4,
    Accent2 : 5,
    Accent3 : 6,
    Accent4 : 7,
    Accent5 : 8,
    Accent6 : 9,
    Hyperlink : 10,
    FollowedHyperlink : 11,
    StyleColor : 12,
};

exports.TickLabelPositionType = {
    High : 0,
    Low : 1,
    NextToAxis : 2,
    None : 3,
};

exports.TickMarkType = {
    Cross : 0,
    Inside : 1,
    None : 2,
    Outside : 3,
};

exports.TotalsCalculation = {
    Sum : 6,
    Count : 2,
    Average : 1,
    Max : 4,
    Min : 5,
    Var : 8,
    CountNums : 3,
    StdDev : 7,
    None : 0,
    Custom : 9,
};

exports.TrendlineType = {
    Exponential : 0,
    Linear : 1,
    Logarithmic : 2,
    MovingAverage : 3,
    Polynomial : 4,
    Power : 5,
};

exports.ValidationAlertType = {
    Information : 0,
    Stop : 1,
    Warning : 2,
};

exports.ValidationType = {
    AnyValue : 0,
    WholeNumber : 1,
    Decimal : 2,
    List : 3,
    Date : 4,
    Time : 5,
    TextLength : 6,
    Custom : 7,
};

exports.ViewType = {
    NormalView : 0,
    PageBreakPreview : 1,
    PageLayoutView : 2,
};

exports.WeightType = {
    HairLine : -1,
    MediumLine : 1,
    SingleLine : 0,
    WideLine : 2,
};

exports.VisibilityType = {
    Visible : 0,
    Hidden : 1,
    VeryHidden : 2,
};

exports.TextParagraphEnumerator.prototype[Symbol.iterator] = function() {
    return this;
};

exports.ReferredAreaEnumerator.prototype[Symbol.iterator] = function() {
    return this;
};

exports.CellEnumerator.prototype[Symbol.iterator] = function() {
    return this;
};

exports.RowEnumerator.prototype[Symbol.iterator] = function() {
    return this;
};

exports.DigitalSignatureEnumerator.prototype[Symbol.iterator] = function() {
    return this;
};

exports.PivotFieldEnumerator.prototype[Symbol.iterator] = function() {
    return this;
};

exports.PivotItemEnumerator.prototype[Symbol.iterator] = function() {
    return this;
};

exports.ChartPointEnumerator.prototype[Symbol.iterator] = function() {
    return this;
};

exports.ExternalLinkEnumerator.prototype[Symbol.iterator] = function() {
    return this;
};
